plugins/arUnogPreservicaPlugin/modules/arUnogPreservicaPlugin/actions/downloadMasterAction.class.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arUnogPreservicaPluginDownloadMasterAction extends sfAction
+{
+    public function execute($request)
+    {
+        $this->resource = $this->getRoute()->resource;
+
+        if (!isset($this->resource)) {
+            $this->forward404();
+        }
+
+        // Attempt to fetch information object's digital object
+        if (empty($do = $this->resource->getDigitalObject())) {
+            $this->forward404();
+        }
+
+        // Make sure user has the right to read the master
+        if (!QubitAcl::check($this->resource, 'readMaster')) {
+            $this->getResponse()->setStatusCode(403);
+
+            return sfView::NONE;
+        }
```

Comment:
Is it worth doing another check if the IO is a draft to make sure the user has permissions, similar to what we did for the API endpoint to download digital objects in https://github.com/artefactual/atom/pull/1234?